src/service-object.ts

```diff
@@ -525,7 +520,9 @@ class ServiceObject<T = any> extends EventEmitter {
    * @param {string} reqOpts.uri - A URI relative to the baseUrl.
    * @param {function} callback - The callback function passed to `request`.
    */
-  request(reqOpts: DecorateRequestOptions): Promise<RequestResponse>;
+  request<T = ResponseBody>(
+    reqOpts: DecorateRequestOptions
+  ): Promise<RequestResponse>;
```

Comment:
Should this be `Promise<RequestResponse<T>>`?

Reply:
I think this is half-finished - I'm going to revert this, and open another PR to genericize the request methods.

Reply:
..as part 3 of #426